File: go/lv0/숫자_찾기/숫자_찾기.go
// golang은 cpp와 다르게 숫자를 문자로 바꾸는 식으로 진행해보자
import "strconv"

func solution(num int, k int) int {
    // num을 문자열로 바꿈
    str := strconv.Itoa(num)
    // k를 문자열로 바꿈
    kStr := strconv.Itoa(k)
    for idx, value := range str {
        // 발견된 첫번째 위치의 값을 리턴해야 하기 때문에 발견되는 즉시 리턴을 해줌
        // 이 때 value는 rune타입이기 때문에 string으로 바꿔줘야 한다.
        if kStr == string(value){
            return idx + 1
        }
    }
    // 일치하는 값이 없으면 그대로 -1 리턴
    return -1
}

File: cpp/lv0/숫자_찾기/숫자_찾기.cpp
#include <string>
#include <vector>

using namespace std;

// 숫자로 먼저 해보자, 계속 10으로 나누고 10나머지 연산을 해서 나머지가 num인 숫자를 찾는다.
// num인 숫자가 있다면 몇번째에 나왔는지 기록해놓고, 나머지연산과 나누기를 게속해서
// 전체 숫자는 몇자리수인지 기록한다.
// 이후 num인 숫자가 있다면, 전체 자리수에서 해당 숫자의 자리수를 빼서 리턴해면 된다.

int solution(int num, int k) {
    if (num == k) {
        return 1;
    }
    int answer = 0;
    int numlength = 0;
    
    while(true) {
        // 전체 숫자 자릿수를 +1한다.
        numlength++;
        if (num % 10 == k) {
            // 조건이 맞으면 k의 위치를 해당 현재 숫자의 길이로 초기화한다.
            answer = numlength;
        }
        num = num / 10;
        
        // 10 이하의 숫자를 10으로 나누면 int는 0이 되기 때문에 break한다.
        if (num == 0) {
            break;
        }
    }
    if (answer == 0) {
        return -1;
    }
    // 뒤에서부터 쟀기 때문에 전체 - 뒤에서부터 발견된 자리 + 1을 해줘야 앞에서부터의 값이 나온다.
    return numlength - answer + 1;
}